phoenix-core/src/it/java/org/apache/phoenix/end2end/join/SubqueryUsingSortMergeJoinIT.java

```diff
@@ -516,8 +516,8 @@ public void testAnyAllComparisonSubquery() throws Exception {
             assertEquals(rs.getString(2), "T6");
 
             assertFalse(rs.next());
-            
-            query = "SELECT /*+ USE_SORT_MERGE_JOIN*/ \"order_id\", name FROM " + tableName4 + " o JOIN " + tableName1 + " i ON o.\"item_id\" = i.\"item_id\" WHERE quantity != ANY(SELECT quantity FROM " + tableName4 + " q WHERE o.\"item_id\" = q.\"item_id\" GROUP BY quantity)";
+            //add order by to make the query result stable
+            query = "SELECT /*+ USE_SORT_MERGE_JOIN*/ \"order_id\", name FROM " + tableName4 + " o JOIN " + tableName1 + " i ON o.\"item_id\" = i.\"item_id\" WHERE quantity != ANY(SELECT quantity FROM " + tableName4 + " q WHERE o.\"item_id\" = q.\"item_id\" GROUP BY quantity) order by \"order_id\"";
```

Comment:
what happens if we don't add the order by?

Reply:
If we don't add the order by , the order of the query result is reverse after this patch, just as:

Reply:
IMO its fine to not have the ```ORDER BY``` and change the test to verify that the order of results is reversed as previously the ```CLIENT SORTED BY [O.ITEM_ID]``` was not needed.

Reply:
@twdsilva , thank you for your opinion, the reason why the query result with this patch is not same as before this patch is different from previous test, if there are more than two tuples which O.ITEM_ID are '0000000006', the query result may not be accurately reversed. Following your suggestion, I would add explain assertions to verify the unnecessary sort is avoid.